plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -251,11 +256,18 @@ protected void copyTableSchema(Connection connection, String catalogName, String
 
             case Types.CHAR:
             case Types.NCHAR:
-                return Optional.of(defaultCharColumnMapping(typeHandle.getRequiredColumnSize()));
+                int columnSize = typeHandle.getRequiredColumnSize();
+                if (columnSize > CharType.MAX_LENGTH) {
+                    return getVarcharType(typeHandle.getRequiredColumnSize());
+                }
+                CharType charType = createCharType(columnSize);
+                // Remote database can be case insensitive.
+                PredicatePushdownController predicatePushdownController = CASE_INSENSITIVE_PUSHDOWN;
+                return Optional.of(ColumnMapping.sliceMapping(charType, charReadFunction(charType), charWriteFunction(), predicatePushdownController));
```

Comment:
This is the same as `io.trino.plugin.jdbc.StandardColumnMappings#defaultCharColumnMapping` except for the case insensitivity. 
I think instead of inlining it in connectors (already 3 of them), we should add a `boolean isRemoteCaseSensitive` parameter to that method.